src/graphql/directives/canMutate.js

```diff
@@ -0,0 +1,48 @@
+import debug from 'debug';
+import { SchemaDirectiveVisitor, ForbiddenError } from 'apollo-server-express';
+import { defaultFieldResolver } from 'graphql';
+import sessionStore from '../../dataSources/cloudFirestore/session';
+import checkMemberCanMutate from '../../lib/checkMemberCanMutate';
+
+const dlog = debug('that:api:sessions:directives:canMutate');
+
+class CanMutateDirective extends SchemaDirectiveVisitor {
+  // eslint-disable-next-line class-methods-use-this
+  visitFieldDefinition(inField) {
+    const field = inField;
+    const { resolve = defaultFieldResolver } = field;
+    // eslint-disable-next-line func-names
+    field.resolve = async function(...args) {
+      dlog('field.resolve called');
+      // dlog('the args:: %O', args);
+      // eslint-disable-next-line no-unused-vars
+      const [id, sessionArg, context] = args;
+      dlog('id %o', id);
+
+      const { user, dataSources } = context;
+      const { firestore } = dataSources;
+
+      let eventId;
+      if (id.sessionId) {
```

Comment:
So I think I would check for both the sessionId and eventId first (as an upfront check) and if they're not there then throw the error.

Reply:
Updated per suggestion in commit a4fd7f0